sw/device/tests/rstmgr_sw_req_test.c

```diff
@@ -0,0 +1,41 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_rstmgr_t rstmgr;
+
+const test_config_t kTestConfig;
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  dif_rstmgr_reset_info_bitfield_t info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &info));
+
+  // If POR, clear reason and request software reset
+  if (info & kDifRstmgrResetInfoPor) {
+    LOG_INFO("POR encountered!");
+    CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+
+    // wait here until device reset
+    wait_for_interrupt();
+  } else if (info & kDifRstmgrResetInfoSw) {
```

Comment:
How about `info == kDifRstmgrResetInfoSw`? Likewise above. We do not expect other bits to be set.

Reply:
sounds good, will do.